lib/pbench/server/templates.py

```diff
@@ -56,6 +75,18 @@ def load(self):
                 raise JsonFileError("{}: {}".format(self.file, err))
         self.json = data
 
+    def merge(self, name: str, base: "JsonFile"):
+        """
+        Base class implementation for merge. The base class has no associated
+        skeleton document to merge but this no-op implementation allows merge
+        to be called on any JsonFile.
```

Comment:
Since this code ostensibly defines the interface for this function (although, it doesn't provide an implementation), it would be nice if the description of the function were more complete.  (E.g., what is an implementation of this function expected to _do_ with the property and the base?  Not having looked at the subclass(es) implementation(s) yet, I'm having a hard time imagining how this is going to work....)

Reply:
I don't understand:  if this function is called, and the caller specifies a base document, the result will be a no-op...won't that come as a surprise to the caller??

I.e., if `base` is `None`, then a no-op result would probably be a good thing; but if `base` is not `None`, then it seems like this method should raise [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError).

Reply:
GitHub is being weird -- it posted my review comments individually, despite giving the appearance that it was saving them up as "a review", so I didn't get a chance to correct them before they posted.

GitHub is being weird -- it posted my review comments individually, despite giving the appearance that it was saving them up as "a review", so I didn't get a chance to correct them before they posted.

> Not having looked at the subclass(es) implementation(s) yet, I'm having a hard time imagining how this is going to work...

I see now that the subclass implementation draws the rest of the information required for the merge from instance attributes, so now the interface makes more sense.  But, it would still be very helpful to have a more complete description here.

Reply:
In fact the final code only calls merge if there's a skeleton document to be merged, so there's no longer any need for this to be a nop on the base class. I'll add a `raise NotImplementedError` to make this clear.